components/script/dom/htmlimageelement.rs

```diff
@@ -414,6 +426,7 @@ impl HTMLImageElement {
 
     // Steps common to when an image has been loaded.
     fn handle_loaded_image(&self, image: Arc<Image>, url: ServoUrl) {
+        self.resolve_image_decode_promises();
```

Comment:
Let's do this at the end of the function, since otherwise the promise handler code will not see the updates happening below.